moveit_core/CMakeLists.txt

```diff
@@ -48,10 +48,13 @@ endif()
 
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(LIBFCL_PC REQUIRED fcl)
-# find *absolute* paths to LIBFCL_INCLUDE_DIRS and LIBFCL_LIBRARIES
-find_path(LIBFCL_INCLUDE_DIRS fcl/config.h HINTS ${LIBFCL_PC_INCLUDE_DIR} ${LIBFCL_PC_INCLUDE_DIRS})
-find_library(LIBFCL_LIBRARIES fcl HINTS ${LIBFCL_PC_LIBRARY_DIRS})
-
+set(LIBFCL_INCLUDE_DIRS ${LIBFCL_PC_INCLUDE_DIRS})
```

Comment:
Newer versions of fcl (ie not the system version in noetic ... what a mess ...) also provide a cmake config. So maybe
```
find_package(fcl REQUIRED)
```
already does the trick?

Reply:
Yes, I tried this before. It works here, locally. But exporting the required `fcl` libs to downstream packages didn't work:
catkin provides two mechanisms to do this: 
1. For catkin packages via `CATKIN_DEPENDS` and 
2. for 3rd-party packages via `DEPENDS` in `catkin_package()`. 

The latter - required here - requires to declare `LIBFCL_INCLUDE_DIRS` and `LIBFCL_LIBRARIES`. But, I failed to assemble `LIBFCL_LIBRARIES` because fcl's cmake config declares an [imported library target](https://cmake.org/cmake/help/latest/command/add_library.html#imported-libraries). Manually accessing [`IMPORTED_LOCATION`](https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_LOCATION.html) and [`INTERFACE_LINK_LIBRARIES`](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_LINK_LIBRARIES.html) of that target was a nightmare and I gave up.

Reply:
Hmm, who besides `collision_detection_fcl` needs that dependency? But looking at the documentation of `catkin_package` I see your problem

Reply:
> Who besides `collision_detection_fcl` needs that dependency?

`moveit_compare_collision_checking_speed_fcl_bullet` was failing to link.

Reply:
Then the clean and lean `find_package` could be moved to these two directly? Normally the fcl dependency should be abstracted away right?

Reply:
Maybe. I'm not sure `moveit_compare_collision_checking_speed_fcl_bullet` explicitly uses fcl functionality. If so, it should `find_package(fcl)` itself, I agree.

Reply:
Given the fact that we need to handle FCL 0.5 for Melodic, we are stuck to pkgconfig for now. 
Hence, I would like to get this basic fix merged as is.